src/shader/AstGen.zig

```diff
@@ -2912,6 +2918,7 @@ fn genGenericFloatTripleBuiltin(
     scope: *Scope,
     node: NodeIndex,
     comptime op: Inst.TripleIntrinsic.Op,
+    comptime scalar_float_arg3: bool,
```

Comment:
@alichraghi - this doesn't handle the case where mix's 3rd argument is a vector, so will need a better solution

Reply:
the builtins analyzations must be definition based as much as possible to avoid these cursed functions. but we can do that later for zsl